lib/infer/kind.js

```diff
@@ -28,30 +27,36 @@ module.exports = function () {
       }
     }
 
-    types.visit(comment.context.ast, {
-      visitClassDeclaration: function () {
+    function visitFunction(path) {
+      if (path.node && path.node.id && path.node.id.name && !!/^[A-Z]/.exec(path.node.id.name)) {
         comment.kind = 'class';
-        this.abort();
-      },
-      visitFunction: function (path) {
-        if (path.value && path.value.id && path.value.id.name && !!/^[A-Z]/.exec(path.value.id.name)) {
-          comment.kind = 'class';
-          this.abort();
-        } else {
-          comment.kind = 'function';
-          this.abort();
-        }
+        path.stop();
+      } else {
+        comment.kind = 'function';
+        path.stop();
+      }
+    }
+
+    if (!comment.context.ast.parentPath) {
+      return comment;
+    }
+
+    comment.context.ast.parentPath.traverse({
+      ClassDeclaration: function (path) {
+        comment.kind = 'class';
+        path.stop();
       },
-      visitTypeAlias: function () {
+      FunctionDeclaration: visitFunction,
+      FunctionExpression: visitFunction,
+      ArrowFunctionExpression: visitFunction,
```

Comment:
If you want you can also visit any node with the `Function` [alias](https://github.com/babel/babel/blob/master/packages/babel-types/src/definitions/es2015.js#L37) including `ClassMethod` and `ObjectMethod` (in addition to what you have)